File: src/services/logService.ts
import type { Log } from "@/schemas";
import { request } from "./api";

/**
 * 任務 (Tasks) 相關 API
 */
export const logService = {
  /**
   * 獲取特定任務的所有日誌。
   * @param taskId - 任務 ID
   */
  getByTaskId(taskId: string): Promise<Log[]> {
    return request<Log[]>('GET', `/api/v1/log/${taskId}`);
  },
};

File: backend/services/logService.py
from backend import schemas
from backend.repositories.log import LogRepository


class LogService:
    def __init__(self, repository: LogRepository):
        self.repository = repository

    def get_logs_by_task_id(self, task_id: str):
        return self.repository.get_by_task_id(task_id)

    def create_log(self, log: schemas.LogCreate):
        return self.repository.create(log)
